src/e2e-test/java/io/cdap/plugin/common/stepsdesign/TestSetupHooks.java

```diff
@@ -1636,4 +1640,12 @@ public static void deleteTargetBqmtTable() throws IOException, InterruptedExcept
       }
     }
   }
+
+  @After(order = 1, value = "@BQ_SECONDARY_RECORD_SOURCE_TEST")
+  public static void deleteTempSource2BQTable() throws IOException, InterruptedException {
+    BigQueryClient.dropBqQuery(bqSourceTable2);
+    PluginPropertyUtils.removePluginProp("bqSourceTable2");
```

Comment:
before removing the plugin property, we need to fetch it for logging right?

Otherwise `bqSourceTable2` would be empty?

Shouldn't it be something like:
```
bqSourceTable2 = PluginPropertyUtils.pluginProp("bqSourceTable2");
PluginPropertyUtils.removePluginProp("bqSourceTable2");
```

Reply:
Actually the logger is at right position. The way you have written is also right but we already assigned globally assigned **bqSourceTable2** in before hook. so no need to add in after hook. And either after removing PluginPropertyUtils.removePluginProp("bqSourceTable2") we still have globally assigned **bqSourceTable2** so we can use logger after remove step.

Reply:
The source of truth here would be the `pluginProperties` file so I think we should always fetch the values from there instead of relying on the before hook.

Reply:
added the step